addons/binding/org.openhab.binding.nikohomecontrol/src/main/java/org/openhab/binding/nikohomecontrol/internal/protocol/NikoHomeControlCommunication.java

```diff
@@ -322,16 +355,18 @@ private void cmdListActions(List<Map<String, String>> data) {
             int id = Integer.parseInt(action.get("id"));
             Integer state = Integer.valueOf(action.get("value1"));
             String value2 = action.get("value2");
+            @SuppressWarnings("null")
             Integer closeTime = (value2 == null ? null : Integer.valueOf(value2));
             String value3 = action.get("value3");
+            @SuppressWarnings("null")
             Integer openTime = (value3 == null ? null : Integer.valueOf(value3));
 
             if (!this.actions.containsKey(id)) {
                 // Initial instantiation of NhcAction class for action object
                 String name = action.get("name");
                 Integer type = Integer.valueOf(action.get("type"));
                 Integer locationId = Integer.valueOf(action.get("location"));
-                String location = null;
+                String location = "";
```

Comment:
Do you want an empty location or should you not even continue to the object construction?

Reply:
Empty location is OK. Null was replaced by an empty string to avoid needing an nullable parameter in the NhcAction constructor call.